udp/connection.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2015 The Chihaya Authors. All rights reserved.
+// Use of this source code is governed by the BSD 2-Clause license,
+// which can be found in the LICENSE file.
+
+package udp
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+)
+
+type ConnectionIDGenerator struct {
+	iv, iv2 []byte
+	block   cipher.Block
+}
+
+func (g *ConnectionIDGenerator) Init() error {
+	key := make([]byte, 16)
+	_, err := rand.Read(key)
+	if err != nil {
+		return err
+	}
+
+	g.block, err = aes.NewCipher(key)
+	if err != nil {
+		return err
+	}
+
+	return g.NewIV()
+}
+
+func (g *ConnectionIDGenerator) Generate(ip []byte) []byte {
+	return g.generate(ip, g.iv)
+}
+
+func (g *ConnectionIDGenerator) generate(ip []byte, iv []byte) []byte {
+	if len(ip) > 16 {
+		panic("IP larger than 16 bytes")
```

Comment:
This should never happen, it's basically an assertion that you didn't accidentally pass in something strange. Actually, we could make the argument type a `net.IP`, would that be better?